libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/AudioCapabilities.java

```diff
@@ -357,6 +365,10 @@ private Api29() {}
     public static int[] getDirectPlaybackSupportedEncodings() {
       ImmutableList.Builder<Integer> supportedEncodingsListBuilder = ImmutableList.builder();
       for (int encoding : ALL_SURROUND_ENCODINGS_AND_MAX_CHANNELS.keySet()) {
+        // Skip ENCODING_DTS_UHD_P2 if API < 34. Otherwise setEncoding will crash.
```

Comment:
I'm thinking of this check is likely to be extended longer as more encodings are added with the platform evolving. How about we create another private static function under Api29 like

```
private static final ArrayList<Integer> getAllSurroundEncodingsMaybeSupported() {
      ArrayList<Integer> encodings = new ArrayList<Integer>();
      for (int encoding : ALL_SURROUND_ENCODINGS_AND_MAX_CHANNELS.keySet()) {
        // AudioFormat.ENCODING_DTS_UHD_P2 is supported from API 34.
        if (Util.SDK_INT >= 34 && encoding == C.ENCODING_DTS_UHD_P2) {
          continue;
        }
        encodings.add(encoding);
      }
      return encodings;
    }
```
And we iterate the array `getAllSurroundEncodingsMaybeSupported()` instead of the current `ALL_SURROUND_ENCODINGS_AND_MAX_CHANNELS` in line 367. This will introduce one more iteration when calling `getAllSurroundEncodingsMaybeSupported()` , considering `ALL_SURROUND_ENCODINGS_AND_MAX_CHANNELS` is in a quite limited size, so should be fine?

Reply:
@tianyif Add getAllSurroundEncodingsMaybeSupported() as suggested. Please see d9129c2e0bebeb930e269408595e2d5da7043062  Thanks.

Reply:
Thanks for the change! Could we return a immutable list instead of the mutable one? Sorry my bad, I should have included this in the above snippet.

Reply:
@tianyif Done. Please see 4a9c804f844dcb062cf1bd1a45021c3f536b78dd. Thanks.